### PR TITLE
Java doc

### PR DESCRIPTION
**Pull request by btkelly**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -19,18 +19,35 @@ allprojects {
 
 dependencies {
     compile gradleApi()
-    compile 'org.codehaus.groovy:groovy-all:2.3.11'
+    compile 'org.codehaus.groovy:groovy-all:2.3.6'
```
Comment (stkent) [inline build.gradle:22]: ?

File: src/main/groovy/com/btkelly/gnag/GnagPlugin.groovy [hunk 2]
```diff
@@ -10,7 +10,7 @@ import org.gradle.api.Project
 import org.gradle.api.invocation.Gradle
 
 /**
- * Created by bobbake4 on 10/23/15.
+ * The main plugin class allowing hooks into the build system to
```
Comment (stkent) [inline src/main/groovy/com/btkelly/gnag/GnagPlugin.groovy]: incomplete?

File: src/main/groovy/com/btkelly/gnag/GnagPluginExtension.groovy [hunk 3]
```diff
@@ -3,35 +3,82 @@ package com.btkelly.gnag
...
+
+    /**
+     * Flag to indicate if the Gnag plugin should make the build as a failure if one of the reporters indicates to do so
+     */
     def boolean failBuildOnError;
 
+    /**
+     * Checks to make sure the current project has the required parameters (gitHubRepoName, gitHubAuthToken, gitHubIssueNumber)
+     * @param project
+     * @return
```
Comment (stkent) [inline src/main/groovy/com/btkelly/gnag/GnagPluginExtension.groovy]: Skipping param and return stuff for now?

File: src/main/groovy/com/btkelly/gnag/reporters/CommentReporter.groovy [hunk 4]
```diff
@@ -3,12 +3,28 @@ package com.btkelly.gnag.reporters
 import org.gradle.api.Project
 
 /**
- * Created by bobbake4 on 10/23/15.
+ * Interface for a comment reporter
  */
 interface CommentReporter {
 
+    /**
+     * This method should inspect the current project and return a true of false if this reporter should cause the build to fail.
```
Comment (stkent) [inline src/main/groovy/com/btkelly/gnag/reporters/CommentReporter.groovy]: true of false -> true or false

**Comment by stkent** [comment]
comments in!

File: build.gradle (hunk 1 above)
Comment (btkelly) [inline build.gradle:22]: There was conflicting versions of the groovy-all dependency so had to down grade it

File: src/main/groovy/com/btkelly/gnag/GnagPlugin.groovy (hunk 2 above)
Comment (btkelly) [inline src/main/groovy/com/btkelly/gnag/GnagPlugin.groovy]: Hahaha good catch

File: src/main/groovy/com/btkelly/gnag/GnagPluginExtension.groovy (hunk 3 above)
Comment (btkelly) [inline src/main/groovy/com/btkelly/gnag/GnagPluginExtension.groovy]: eh I did it in some places, I guess it should be consistent

File: src/main/groovy/com/btkelly/gnag/reporters/CommentReporter.groovy (hunk 4 above)
Comment (btkelly) [inline src/main/groovy/com/btkelly/gnag/reporters/CommentReporter.groovy]: yup

File: build.gradle (hunk 1 above)
Comment (stkent) [inline build.gradle:22]: :+1:

**Pull request merged**